source/plugin/blocks/accessing.rst

```diff
@@ -34,56 +34,128 @@ referenced block is any kind of banner by checking the blocks type.
 
 .. code-block:: java
 
     import org.spongepowered.api.block.BlockState;
-    import org.spongepowered.api.data.manipulator.WetData;
+    import org.spongepowered.api.data.manipulator.immutable.ImmutableWetData;
 
-    public boolean isWet(Location block) {
-        if (block.getBlock().getType() != BlockTypes.SPONGE) {
+    public void isWet(Location blockLoc) {
+        BlockState sponge = blockLoc.getBlock();
+        if (!sponge.getType().equals(BlockTypes.SPONGE)) {
             return false;
         }
-        BlockState sponge = block.getBlock();
-        Optional<WetData> wetness = sponge.getManipulator(WetData.class);
+        Optional<ImmutableWetData> wetness = sponge.get(ImmutableWetData.class);
         return wetness.isPresent();
     }
 
-First, we need to know which ``DataManipulator`` subinterface we need. Those that are applicable to blocks are found in
-the ``org.spongepowered.api.data.manipulator`` and ``org.spongepowered.api.data.manipulator.block`` packages. Then we
-can just pass that class to the ``getManipulator()`` method and get an ``Optional`` in return which will be ``empty()``
-if the ``BlockState`` does not contain data of that type. Since ``WetData`` represents a boolean value, its presence
-equates to ``true``. Its absence (if ``Optional.empty()`` was returned) either signifies ``false``.
+More information on mutable and immutable ``DataManipulator`` s can be found in the :doc:`data documentation <../data/datamanipulators>`.
+
+Block Properties
+~~~~~~~~~~~~~~~~
+Blocks can contain certain properties. A property is a pre-set value that defines the game logic of that particular block. For example, blocks can contain pre-determined blast-resistance values that can be used to determine what you're working with, without actually checking the type of block it could be one by one. For example, if we wanted to get the blast resistance of a block and checking if it is greater than or equal to one, it would be done like so:
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.property.DoubleProperty;
+    import org.spongepowered.api.data.property.block.BlastResistanceProperty;
+
+    public boolean blastResistanceGreaterThanOne(Location<World> blockLoc) {
+        Optional<BlastResistanceProperty> optional =
+            blockLoc.getProperty(BlastResistanceProperty.class);
+        
+        if(optional.isPresent()) {
+            BlastResistanceProperty resistance = optional.get();
+            return resistance.matches(DoubleProperty.greaterThanOrEqual(1));
+        }
+        return false;
+    }
+
+What this will do, is it will get the blast resistance of our block and compare it to a new ``DoubleProperty``, as ``BlastResistantProperty`` is a sub-interface of ``DoubleProperty``. The method will then return if the blast resistance of our block is greater than one, the value in placed ``matches()``. If we wanted to see if it was less than two, we would replace it with ``lessThan()``.
```

Comment:
You may want to say that `BlastResistanceProperty` **inherits from** `DoubleProperty`